src/IRsend.cpp

```diff
@@ -340,7 +340,12 @@ void IRsend::sendGeneric(const uint16_t headermark, const uint32_t headerspace,
 
     // Footer
     if (footermark)  mark(footermark);
-    space(std::max(gap, mesgtime - usecs.elapsed()));
+    uint32_t elapsed = usecs.elapsed();
+    // Avoid potential unsigned integer underflow. e.g. when mesgtime is 0.
+    if (elapsed >= mesgtime)
+      space(gap);
+    else
+      space(std::max(gap, mesgtime - elapsed));
```

Comment:
you still avoid duplicating the rest of the function so i think its worth it.
if you don't like the if statement another way is using one line if statement like this:

uint32_t space_gap = elapsed > mesgtime ? gap : std::max(gap, mesgtime - elapsed);
space(space_gap);

Reply:
Agreed. No doubt the de-duplication of the procedure was worth it. Just the road to heck is/was paved with good intentions. ;-)

I don't think changing the code to the one-liner actually saves any compiled code/space. The compiler should do that the optimal anyway. If it does save space, I'll change it. However I think the code is more readable to others with a `if` statement verses the conditional assignment.